HLTrigger/JetMET/interface/HLTHFJetCleaner.h

```diff
@@ -0,0 +1,224 @@
+
+  edm::EDGetTokenT<edm::ValueMap<float>> sigmaEtaEtaToken_;
+  edm::EDGetTokenT<edm::ValueMap<float>> sigmaPhiPhiToken_;
+  edm::EDGetTokenT<edm::ValueMap<int>> centralEtaStripSizeToken_;
+
+  // Minimum jet pt to consider while applying HF cuts
+  const float jetPtThresh_;
+  const float dphiJetMetThresh_;
+
+  // Cuts will be applied to jets only with jetEtaMin_ < |eta| < jetEtaMax_
+  const float jetEtaMin_;
+  const float jetEtaMax_;
+
+  const float sigmaEtaPhiDiffThresh_;
+  const float centralEtaStripSizeThresh_;
+  const float cornerCutSigmaEtaEta_;
+  const float cornerCutSigmaPhiPhi_;
+  const bool applyCornerCut_;
+  const bool applySigmaEtaPhiCut_;
+  const bool applyStripSizeCut_;
+};
+
+//
+// constructors and destructor
+//
+template <typename JetType>
+HLTHFJetCleaner<JetType>::HLTHFJetCleaner(const edm::ParameterSet& iConfig)
+    : jetToken_(consumes<std::vector<JetType>>(iConfig.getParameter<edm::InputTag>("jets"))),
+      metToken_(consumes<edm::View<reco::MET>>(iConfig.getParameter<edm::InputTag>("mets"))),
+      sigmaEtaEtaToken_(consumes<edm::ValueMap<float>>(iConfig.getParameter<edm::InputTag>("sigmaEtaEta"))),
+      sigmaPhiPhiToken_(consumes<edm::ValueMap<float>>(iConfig.getParameter<edm::InputTag>("sigmaPhiPhi"))),
+      centralEtaStripSizeToken_(
+          consumes<edm::ValueMap<int>>(iConfig.getParameter<edm::InputTag>("centralEtaStripSize"))),
+      jetPtThresh_(iConfig.getParameter<double>("jetPtThresh")),
+      dphiJetMetThresh_(iConfig.getParameter<double>("dphiJetMetThresh")),
+      jetEtaMin_(iConfig.getParameter<double>("jetEtaMin")),
+      jetEtaMax_(iConfig.getParameter<double>("jetEtaMax")),
+      sigmaEtaPhiDiffThresh_(iConfig.getParameter<double>("sigmaEtaPhiDiffThresh")),
+      centralEtaStripSizeThresh_(iConfig.getParameter<int>("centralEtaStripSizeThresh")),
+      cornerCutSigmaEtaEta_(iConfig.getParameter<double>("cornerCutSigmaEtaEta")),
+      cornerCutSigmaPhiPhi_(iConfig.getParameter<double>("cornerCutSigmaPhiPhi")),
+      applyCornerCut_(iConfig.getParameter<bool>("applyCornerCut")),
+      applySigmaEtaPhiCut_(iConfig.getParameter<bool>("applySigmaEtaPhiCut")),
+      applyStripSizeCut_(iConfig.getParameter<bool>("applyStripSizeCut")) {
+  produces<JetCollection>();
+}
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+template <typename JetType>
+void HLTHFJetCleaner<JetType>::produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const {
+  edm::Handle<std::vector<JetType>> jets;
+  edm::Handle<edm::View<reco::MET>> mets;
+
+  edm::Handle<edm::ValueMap<float>> sigmaEtaEtas;
+  edm::Handle<edm::ValueMap<float>> sigmaPhiPhis;
+  edm::Handle<edm::ValueMap<int>> centralEtaStripSizes;
+
+  iEvent.getByToken(jetToken_, jets);
+
+  iEvent.getByToken(sigmaEtaEtaToken_, sigmaEtaEtas);
+  iEvent.getByToken(sigmaPhiPhiToken_, sigmaPhiPhis);
+  iEvent.getByToken(centralEtaStripSizeToken_, centralEtaStripSizes);
```

Comment:
When the handle is needed use `iEvent.getHandle(token)`. When the Handle is not needed, use `iEvent.get(token)`: [example](https://github.com/cms-sw/cmssw/blob/3ca0032ab77b75be849b9a3a603a52e6d16ebdf0/HLTrigger/JetMET/plugins/HLTJetTimingFilter.h#L68).